cmd/ulxly/ulxly.go

```diff
@@ -126,6 +182,294 @@ var DepositsCmd = &cobra.Command{
 	},
 }
 
+//go:embed depositNewUsage.md
+var depositNewUsage string
+var depositNewCmd = &cobra.Command{
+	Use:     "deposit-new",
+	Short:   "Make a uLxLy deposit transaction",
+	Long:    depositNewUsage,
+	Args:    cobra.NoArgs,
+	PreRunE: checkDepositArgs,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		ctx := cmd.Context()
+		// Dial the Ethereum RPC server.
+		client, err := ethclient.DialContext(ctx, *ulxlyInputArgs.DepositRPCURL)
+		if err != nil {
+			log.Error().Err(err).Msg("Unable to dial rpc")
+		}
+		defer client.Close()
+
+		bridgeV2, err := ulxly.NewUlxly(common.HexToAddress(*ulxlyInputArgs.DepositBridgeAddress), client)
+		if err != nil {
+			return err
+		}
+
+		privateKey, err := crypto.HexToECDSA(*ulxlyInputArgs.PrivateKey)
+		if err != nil {
+			log.Error().Err(err).Msg("Unable to retrieve private key")
+		}
+
+		publicKey := privateKey.Public()
+		publicKeyECDSA, ok := publicKey.(*ecdsa.PublicKey)
+		if !ok {
+			log.Error().Msg("Error casting public key to ECDSA")
+		}
+
+		fromAddress := crypto.PubkeyToAddress(*publicKeyECDSA)
+
+		value := big.NewInt(*ulxlyInputArgs.Value)
+		gasLimit := *ulxlyInputArgs.GasLimit // Deposit transactions seem to use >300,000 units of gas to be safe.
+		gasPrice, err := client.SuggestGasPrice(ctx)
+		if err != nil {
+			log.Error().Err(err).Msg("Cannot get suggested gas price")
+		}
+		// gasTipCap, err := client.SuggestGasTipCap(ctx)
+		// if err != nil {
+		// 	log.Error().Err(err).Msg("Cannot get suggested gas tip cap")
+		// }
+
+		toAddress := common.HexToAddress(*ulxlyInputArgs.DestinationAddress)
+		tokenAddress := common.HexToAddress(*ulxlyInputArgs.TokenAddress)
+		metaBytes := common.Hex2Bytes(*ulxlyInputArgs.MetaBytes)
+
+		chainID, err := client.NetworkID(ctx)
+		if err != nil {
+			log.Error().Err(err).Msg("Cannot get chain ID")
+		}
+
+		signer := types.LatestSignerForChainID(chainID)
+
+		var tops *bind.TransactOpts
+		if tokenAddress == common.HexToAddress("0x0000000000000000000000000000000000000000") {
+			tops = &bind.TransactOpts{
+				Signer: func(address common.Address, transaction *types.Transaction) (*types.Transaction, error) {
+					return types.SignTx(transaction, signer, privateKey)
+				},
+				Value:     value,
+				From:      fromAddress,
+				Context:   ctx,
+				GasLimit:  gasLimit,
+				GasPrice:  gasPrice,
+				GasFeeCap: nil,
+				GasTipCap: nil,
+			}
+		} else {
+			tops = &bind.TransactOpts{
+				Signer: func(address common.Address, transaction *types.Transaction) (*types.Transaction, error) {
+					return types.SignTx(transaction, signer, privateKey)
+				},
+				From:      fromAddress,
+				Context:   ctx,
+				GasLimit:  gasLimit,
+				GasPrice:  gasPrice,
+				GasFeeCap: nil,
+				GasTipCap: nil,
+			}
+		}
```

Comment:
I generally prefer to avoid `else` unless it'd needed. If you anticipate that most of the time won't have a token address, you could also switch.. But this is just a style thing.
```suggestion
                tops := &bind.TransactOpts{
                        Signer: func(address common.Address, transaction *types.Transaction) (*types.Transaction, error) {
                                return types.SignTx(transaction, signer, privateKey)
                        },
                        From:      fromAddress,
                        Context:   ctx,
                        GasLimit:  gasLimit,
                        GasPrice:  gasPrice,
                        GasFeeCap: nil,
                        GasTipCap: nil,
                }
		if tokenAddress == common.HexToAddress("0x0000000000000000000000000000000000000000") {
			tops = &bind.TransactOpts{
				Signer: func(address common.Address, transaction *types.Transaction) (*types.Transaction, error) {
					return types.SignTx(transaction, signer, privateKey)
				},
				Value:     value,
				From:      fromAddress,
				Context:   ctx,
				GasLimit:  gasLimit,
				GasPrice:  gasPrice,
				GasFeeCap: nil,
				GasTipCap: nil,
			}
		}
```